From 856def82314965e36e93cf9aab31e721f78b284e Mon Sep 17 00:00:00 2001
From: robinz1x <robinx.zhang@intel.com>
Date: Fri, 1 Jun 2018 14:30:27 +0800
Subject: [PATCH 04/20] Check whether "mNavigationBarController" is null before
 system use it.

[Port from broxton]
"mNavigationBarController" will not be initialized in CarStatusBar always.
So it is always null while system call it in "LightBarController".
In car device, we shouldn't use "mNavigationBarController", so we should
check whether it is null before system use it.

Change-Id: I4fae45f40ae754e1775ec3a5dfab139f3729c78b
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-63680
Signed-off-by: robinz1x <robinx.zhang@intel.com>
---
 .../com/android/systemui/statusbar/phone/LightBarController.java  | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/LightBarController.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/LightBarController.java
index 533771a..feb7dfc 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/LightBarController.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/LightBarController.java
@@ -264,8 +264,10 @@ public class LightBarController implements BatteryController.BatteryStateChangeC
         pw.println(" StatusBarTransitionsController:");
         mStatusBarIconController.getTransitionsController().dump(fd, pw, args);
         pw.println();
-        pw.println(" NavigationBarTransitionsController:");
-        mNavigationBarController.dump(fd, pw, args);
-        pw.println();
+        if(mNavigationBarController != null) {
+            pw.println(" NavigationBarTransitionsController:");
+            mNavigationBarController.dump(fd, pw, args);
+            pw.println();
+        }
     }
 }
-- 
1.9.1

